engine/setup/src/include/general.cpp

```diff
@@ -299,10 +382,14 @@ void SetString(GString **ptr, char *line) {
 
 #endif
 
-char *NewString(const char *line) {
+char *NewString(const char *line) 
```

Comment:
We should check the usages and add a size_t length parameter too. Ideally though this would be a C++ class.